### PR TITLE
Restyle left-nav bar

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Restyle left-nav bar.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/18980
 
 
Screenshot from 2021-04-08 14-38-47

### CODE REVIEW

**Review by sleshchenko — COMMENTED** [review]
Tested and seems to be as expected.

**Review by sleshchenko — APPROVED** [review]

**Review by parvathyvr — APPROVED** [review]
LGTM!

File: src/Layout/Navigation/MainList.tsx [hunk 1]
```diff
@@ -36,7 +29,7 @@ function NavigationMainList(props: { activePath: string }): React.ReactElement {
         item={item}
         activePath={props.activePath}
       >
-        {item.icon}
+        {item.icon && (item.icon)}
```
Comment (akurinnoy) [inline src/Layout/Navigation/MainList.tsx:32]: Why don't you remove this line?

File: src/Layout/Navigation/RecentList.tsx [hunk 2]
```diff
@@ -15,31 +15,13 @@ import {
...
-import styles from './index.module.css';
-import { buildGettingStartedPath } from '../../services/helpers/location';
 import { History } from 'history';
 
-function buildCreateWorkspaceItem(): React.ReactElement {
-  const createWorkspacePath = buildGettingStartedPath('custom-workspace');
-  const item: NavigationItemObject = {
-    to: createWorkspacePath,
-    label: 'Create Workspace',
-    icon: <PlusIcon className={styles.mainItemIcon} />,
```
Comment (akurinnoy) [inline src/Layout/Navigation/RecentList.tsx:34]: please clean up this rule from the corresponding CSS module.

File: src/Layout/Navigation/MainList.tsx (hunk 1 above)
Comment (olexii4) [reply src/Layout/Navigation/MainList.tsx:32]: I thought about the way to return all icons back. I can remove it.

File: src/Layout/Navigation/RecentList.tsx (hunk 2 above)
Comment (olexii4) [reply src/Layout/Navigation/RecentList.tsx:34]: I will do it

**Review by JPinkney — APPROVED** [review]

File: src/Layout/Navigation/__tests__/MainList.spec.tsx [hunk 3]
```diff
@@ -11,41 +11,57 @@
...
+  it('should have correct navigation item workspaces quantity', () => {
+    let workspaces = [0, 1, 2, 3, 4].map(i => createFakeWorkspace('works-' + i, 'works-' + i));
+    const { rerender } = render(buildElement(workspaces));
+
+    expect(screen.getAllByRole('link')[1]).toHaveTextContent('Workspaces (5)');
+
+    workspaces = [0, 1, 2].map(i => createFakeWorkspace('works-' + i, 'works-' + i));
+    rerender(buildElement(workspaces));
+
+    expect(screen.getAllByRole('link')[1]).toHaveTextContent('Workspaces (3)');
```
Comment (akurinnoy) [inline src/Layout/Navigation/__tests__/MainList.spec.tsx:50]:
```suggestion
    expect(screen.queryByRole('link', { name: 'Workspaces (5)' })).toBeInTheDocument();

    workspaces = [0, 1, 2].map(i => createFakeWorkspace('works-' + i, 'works-' + i));
    rerender(buildElement(workspaces));

    expect(screen.queryByRole('link', { name: 'Workspaces (3)' })).toBeInTheDocument();
```

**Review by akurinnoy — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@oorel: it would be really nice if Happy path tests failure wouldn't be ignored next time https://github.com/eclipse/che-dashboard/pull/208#issuecomment-816124086
Now we Happy path tests check blocker in Eclipse Che repos: https://github.com/eclipse/che/issues/19557

**Pull request merged**